File: xide-standard-plugins/xide-settings-plugin/src/main/java/org/xast/xide/settings_plugin/SettingsTool.java
package org.xast.xide.settings_plugin;

import javax.swing.JFrame;

import org.xast.xide.core.plugin.tool.Tool;

public class SettingsTool implements Tool {
    private final JFrame frame;

    public SettingsTool(JFrame frame) {
        this.frame = frame;
    }

    @Override
    public void show() {
        new SettingsDialog(frame)
            .setVisible(true);
    }
}

File: xide-standard-plugins/xide-settings-plugin/src/main/scala/org/xast/xide/settings_plugin/SettingsTool.scala
package org.xast.xide.settings_plugin

import javax.swing.JFrame
import org.xast.xide.core.plugin.tool.Tool

class SettingsTool(val frame: JFrame) extends Tool {
    override def show(): Unit = {
        new SettingsDialog(this.frame).setVisible(true)
    }
}
